lte/gateway/dev_tools.py

```diff
@@ -142,14 +179,28 @@ def _register_network(network_type: str, payload: Any):
     if not dev_utils.does_network_exist(network_id):
         dev_utils.cloud_post(network_type, payload)
 
-    dev_utils.create_tier_if_not_exists(network_id, 'default')
 
+def _register_agw(
+        network_type: str,
+        url: str,
+        admin_cert: Optional[types.ClientCert],
+        network_id: Optional[str],
+):
+    network_id = network_id or NIDS_BY_TYPE[network_type]
+
+    dev_utils.create_tier_if_not_exists(
```

Comment:
So this function explicitly creates an AGW (`lte/:network_id:/gateway`), independent of the network type

But it's the same story for both AGWs and FeGs -- they require an underlying tier to exist https://github.com/magma/magma/blob/abab96b287980f7874bc03434cc2fa9795363d7f/orc8r/cloud/go/services/orchestrator/obsidian/handlers/gateway_handlers.go#L175-L182

So in this case, it should be safe to do it this way -- because if the tier didn't exist, registering the gateway would fail

Reply:
Taking into consideration Tier is a network property (1 tier N Gateways), and  that a gateway shouldn't exist without a network, those changes seems to allow that.

In other words, trying to create a gateway without a network or a tier should raise an error, and this script is hiding this. 

Eventually you need to create a tier, either you do when you create the network or you create the gateway. Since this is a one time thing, this should be part of the network creation. 

But to me the most important reason is,  his is a script that doesn't have any way to test its reliability. Is this change really achieving something worth that risk?  Do you have the time to test the different permutations of adding and removing gateways you will find at fabscripts at `~/magma/lte/gateway/python/integ_tests/federated_tests` and `~/magma/lte/gateway/'